mono/metadata/sre.c

```diff
@@ -1547,7 +1549,21 @@ mono_reflection_type_get_handle (MonoReflectionType* ref, MonoError *error)
 				return NULL;
 			}
 		}
-
+		/* Need to resolve the generic_type in order for it to create its generic context. */
```

Comment:
It kinda looks like avoiding half constructed types might be impossible to do. So these changes are ok.